crates/distribution-types/src/resolution.rs

```diff
@@ -56,99 +57,76 @@ impl Resolution {
         self.0.is_empty()
     }
 
-    /// Return the set of [`Requirement`]s that this resolution represents, exclusive of any
+    /// Return the set of [`UvRequirement`]s that this resolution represents, exclusive of any
     /// editable requirements.
-    pub fn requirements(&self) -> Vec<Requirement> {
+    pub fn requirements(&self) -> Vec<UvRequirement> {
         let mut requirements = self
             .0
             .values()
             // Remove editable requirements
             .filter(|dist| !dist.is_editable())
-            .map(|dist| Requirement::from(dist.clone()))
+            .map(dist_to_uv_requirement)
             .collect::<Vec<_>>();
         requirements.sort_unstable_by(|a, b| a.name.cmp(&b.name));
         requirements
     }
 }
 
-impl From<Dist> for Requirement {
-    fn from(dist: Dist) -> Self {
-        match dist {
-            Dist::Built(BuiltDist::Registry(wheel)) => Self {
-                name: wheel.filename.name,
-                extras: vec![],
-                version_or_url: Some(pep508_rs::VersionOrUrl::VersionSpecifier(
-                    pep440_rs::VersionSpecifiers::from(
-                        pep440_rs::VersionSpecifier::equals_version(wheel.filename.version),
-                    ),
-                )),
-                marker: None,
+fn dist_to_uv_requirement(resolved_dist: &ResolvedDist) -> UvRequirement {
```

Comment:
Should this be `impl From<&ResolvedDist> for Requirement`?

Reply:
done